### ISSUE TITLE
Rule proposal: Forbid range `Z-a`

### ISSUE THREAD

**Issue by fisker** [ota-meshi/eslint-plugin-regexp#138]
New users may use `/[A-f]/`, to match `[A-Za-f]` or `[A-Fa-F]`, but it's neither, there are other strings in this range, they just don't know.

```js
/[A-f]/.test('[')
true
```

### Fail

```js
/[A-z]/
```

Any range include `Z-a`

### Pass

```js
/[A-Za-z]/
```

**Comment by fisker** [comment]
Maybe also `9-A`

**Comment by RunDevelopment** [comment]
#122 does exactly this :)

**Comment by fisker** [comment]
Thanks!
